src/c_api.rs

```diff
@@ -0,0 +1,612 @@
+use std::ffi::CString;
+use std::os::raw::{c_double, c_int, c_long};
+use std::ptr::null_mut;
+use std::str::FromStr;
+use std::{
+    ffi::CStr,
+    os::raw::{c_char, c_void},
+};
+
+use crate::redisjson::Format;
+use crate::{redisjson::RedisJSON, REDIS_JSON_TYPE};
+use redis_module::key::RedisKeyWritable;
+use redis_module::logging::log_notice;
+use redis_module::Context;
+use redis_module::{raw as rawmod, RedisError};
+use serde_json::Value;
+
+// extern crate readies_wd40;
+// use crate::readies_wd40::{BB, _BB, getenv};
+
+//
+// structs
+//
+
+#[repr(C)]
+#[derive(Debug, Copy, Clone)]
+pub struct RedisModuleCtx {
+    _unused: [u8; 0],
+}
+
+#[repr(C)]
+pub enum JSONType {
+    String = 0,
+    Int = 1,
+    Double = 2,
+    Bool = 3,
+    Object = 4,
+    Array = 5,
+    Null = 6,
+}
+
+#[repr(C)]
+#[allow(non_camel_case_types)]
+pub enum RedisReturnCode {
+    REDISMODULE_OK = 0,
+    REDISMODULE_ERR = 1,
+}
```

Comment:
As above -- we already have this in `redismodule-rs`.

Reply:
`RedisReturnCode` was newly introduced here 😊 

`REDISMODULE_OK` and `REDISMODULE_ERR` were already used in `redismodule-rs`
in `enum Status`
The values were generated by `bindgen` from `redismodule.h`
```
pub const REDISMODULE_OK: isize = 0;
pub const REDISMODULE_ERR: isize = 1;
```
Will use `Status` and get rid of `RedisReturnCode`